src/Resources/public/js/search.js

```diff
@@ -43,43 +43,115 @@ class SearchManager {
             options
         );
 
+        // Dynamically bind all functions in options to `this`
+        Object.entries(this.#options).forEach(([key, value]) => {
+            if (typeof value === 'function') {
+                this.#options[key] = value.bind(this);
+            }
+        });
+
+        this.#initializeForm();
+    }
+
+    #initializeForm() {
         let form = this.#options.form;
 
-        if(typeof form === 'string') {
+        if (typeof form === 'string') {
             form = document.querySelector(form);
 
-            if(null === form) {
+            if (null === form) {
                 throw new Error('Form not found');
             }
         }
 
         this.#form = form;
 
-        this.#form.addEventListener('input', function (event) {
-            /** @type {HTMLInputElement} */
+        this.#form.addEventListener('input', (event) => {
             const field = event.target;
 
+            if (!(field instanceof HTMLInputElement || field instanceof HTMLSelectElement)) {
+                return;
+            }
+
             field.dispatchEvent(new CustomEvent('search:form-changed', { bubbles: true }));
 
             // todo would be better to have a way to check the type of the field directly on the field, e.g. with a data attribute
-            if(field.closest('.ssm-filter') !== null) {
+            if (field.closest('.ssm-filter') !== null) {
                 field.dispatchEvent(new CustomEvent('search:filter-changed', { bubbles: true }));
-            } else if(field.name === 'p') {
+            } else if (field.name === 'p') {
                 field.dispatchEvent(new CustomEvent('search:page-changed', { bubbles: true }));
-            } else if(field.classList.contains('ssm-sort')) {
+            } else if (field.classList.contains('ssm-sort')) {
                 field.dispatchEvent(new CustomEvent('search:sort-changed', { bubbles: true }));
             }
         });
 
-        this.#form.addEventListener('search:filter-changed', (event) => this.#options.onFilterChange.bind(this, this.#form, event.target)());
-        this.#form.addEventListener('search:page-changed', (event) => this.#options.onSortChange.bind(this, this.#form, event.target)());
-        this.#form.addEventListener('search:sort-changed', (event) => this.#options.onSortChange.bind(this, this.#form, event.target)());
-        this.#form.addEventListener('submit', () => this.#options.onSubmit.bind(this, this.#form)());
+        this.#form.addEventListener('search:filter-changed', (event) => this.#options.onFilterChange(this.#form, event.target));
+        this.#form.addEventListener('search:page-changed', (event) => this.#options.onPageChange(this.#form, event.target));
+        this.#form.addEventListener('search:sort-changed', (event) => this.#options.onSortChange(this.#form, event.target));
+        this.#form.addEventListener('submit', (event) => {
+            event.preventDefault();
+            this.#updateURL();
```

Comment:
This line doesn't belong here, does it?

Shouldn't it be in the defined callback or in the `#submitForm` method?

Reply:
Maybe it shouldn't be here but I put it here to avoid it being accidentally left out when the callback is overridden.

Reply:
That's a good point. Can we make it cleaner?

Reply:
Like making another listener, which only responsibility is to update the URL?

Reply:
Yeah, that will be better.